torch/quantization/fx/graph_module.py

```diff
@@ -47,8 +44,33 @@ def __deepcopy__(self, memo):
         fake_mod.__dict__ = copy.deepcopy(self.__dict__)
         return ObservedStandaloneGraphModule(fake_mod, self.graph)
 
-def mark_observed_standalone_module(module: GraphModule) -> GraphModule:
-    return ObservedStandaloneGraphModule(module, module.graph)
-
 def is_observed_standalone_module(module: Any) -> bool:
     return isinstance(module, ObservedStandaloneGraphModule)
+
+
+def _save_packed_weight(self, destination, prefix, keep_vars):
+    for attr_name in dir(self):
+        if "_packed_weight" in attr_name and \
+           isinstance(getattr(self, attr_name), torch._C.ScriptObject):  # type: ignore
+            packed_weight = getattr(self, attr_name)
+            destination[prefix + attr_name] = packed_weight
+
+class QuantizedGraphModule(GraphModule):
+    def __init__(self, root: Union[torch.nn.Module, Dict[str, Any]], graph: Graph):
+        super().__init__(root, graph)
+        self._register_state_dict_hook(_save_packed_weight)
```

Comment:
just checking, would this work if the module is scripted?

Reply:
I don't think hooks are supported for script modules. cc @albanD to confirm

Reply:
I don't know about state dict ones. But regular hooks are not supported, so I guess not.

Reply:
ah, I see.  Would it make sense to add some test cases for making sure the state dict loading and saving works after scripting, just to be safe?

Reply:
raghu had a similar comment above, but I remember from james/michael that state_dict is not a supported api in script modules